diff --git a/test.sh b/test.sh
index 8d48d84..7d879df 100755
--- a/test.sh
+++ b/test.sh
@@ -23,7 +23,7 @@ echo "Testing $sha - $desc" | tee $logfile
 mkdir -p $DIR
 cd $DIR
 echo hi from `pwd` >>$logfile
-cmake -G "Ninja" -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=`pwd`/../install  ../dealii >>$logfile 2>&1 && nice ninja install >>$logfile 2>&1 || exit -1
+cmake -G "Ninja" -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=`pwd`/../install  ../dealii/deal.II >>$logfile 2>&1 && nice ninja install >>$logfile 2>&1 || exit -1
 
 cd ..
 
