diff --git a/step-22/step-22.cc b/step-22/step-22.cc
index 59f3a23..da58964 100644
--- a/step-22/step-22.cc
+++ b/step-22/step-22.cc
@@ -1287,7 +1287,7 @@ namespace Step22
          cell != triangulation.end(); ++cell)
       for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
         if (cell->face(f)->center()[dim-1] == 0)
-          cell->face(f)->set_all_boundary_ids(1);
+          cell->face(f)->set_all_boundary_indicators(1);
 
 
     // We then apply an initial refinement
diff --git a/test_assembly/step-22.cc b/test_assembly/step-22.cc
index 53ea14f..47167c3 100644
--- a/test_assembly/step-22.cc
+++ b/test_assembly/step-22.cc
@@ -697,7 +697,7 @@ namespace Step22
 
       Instrument ii1("setup:make_sp");
       timer.enter_section("make dsp");
-      BlockDynamicSparsityPattern dsp (2,2);
+      BlockCompressedSimpleSparsityPattern dsp (2,2);
 
       dsp.block(0,0).reinit (n_u, n_u);
       dsp.block(1,0).reinit (n_p, n_u);
@@ -1265,7 +1265,7 @@ namespace Step22
          cell != triangulation.end(); ++cell)
       for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
         if (cell->face(f)->center()[dim-1] == 0)
-          cell->face(f)->set_all_boundary_ids(1);
+          cell->face(f)->set_all_boundary_indicators(1);
 
 
     // We then apply an initial refinement
